blog/_posts/2018-05-02-developing-on-kubernetes.md

```diff
@@ -0,0 +1,366 @@
+$ ksync get -n dok
+```
+
+Now we deploy the stock generator and the stock consumer microservice:
+
+```
+$ kubectl -n=dok apply \
+      -f https://raw.githubusercontent.com/kubernauts/dok-example-us/master/stock-gen/app.yaml
+$ kubectl -n=dok apply \
+      -f https://raw.githubusercontent.com/kubernauts/dok-example-us/master/stock-con/app.yaml
+```
+
+Once both deployments are created and the pods are running, we forward the `stock-con` service for local consumption (in a separate terminal session):
+
+```
+$ kubectl get -n dok po --selector=app=stock-con  \
+                     -o=custom-columns=:metadata.name --no-headers |  \
+                     xargs -IPOD kubectl -n dok port-forward POD 9898:9898
+```
+
+With that we should be able to consume the `stock-con` service from our local machine; we do this by regularly checking the response of the `healthz` endpoint like so (in a separate terminal session):
+
+```
+$ watch curl localhost:9898/healthz
+```
+
+Now change the code in the `ksync/stock-con`directory, for example update the [/healthz endpoint code in service.js](https://github.com/kubernauts/dok-example-us/blob/2334ee8fb11f8813370122bd46285cf45bdd4c48/stock-con/service.js#L52) by adding a field to the JSON response and observe how the pod gets updated and the response of the `curl localhost:9898/healthz` command changes. Overall you should have something like the following in the end:
+
+![Preview](/images/blog-logging/2018-04-25-developing-on-kubernetes/dok-ksync_preview.png)
+
+### Walkthrough: Minikube with local build
+
+For the following you will need to have Minikube up and running and we will leverage the Minikube-internal Docker daemon for building images, locally. As a preparation, do the following
+
+```
+$ git clone https://github.com/kubernauts/dok-example-us.git && cd dok-example-us
+$ eval $(minikube docker-env)
+$ kubectl create namespace dok
+```
+
+Now we deploy the stock generator and the stock consumer microservice:
+
+```
+$ kubectl -n=dok apply -f stock-gen/app.yaml
+$ kubectl -n=dok apply -f stock-con/app.yaml
+```
+Once both deployments are created and the pods are running, we forward the `stock-con` service for local consumption (in a separate terminal session) and check the response of the `healthz` endpoint:
+
+```
+$ kubectl get -n dok po --selector=app=stock-con  \
+                     -o=custom-columns=:metadata.name --no-headers |  \
+                     xargs -IPOD kubectl -n dok port-forward POD 9898:9898 &
+$ watch curl localhost:9898/healthz
+```
+
+Now change the code in the `stock-con`directory, for example, update the []/healthz endpoint code in service.js](https://github.com/kubernauts/dok-example-us/blob/2334ee8fb11f8813370122bd46285cf45bdd4c48/stock-con/service.js#L52) by adding a field to the JSON response. Once you’re done with your code update, the last step is to build a new container image and kick off a new deployment like shown below:
```

Comment:
the link may not work

Reply:
This is a GitHub permalink. The whole purpose of the link is that it is stable since it references a concrete commit rather than, say, master. Can you please clarify what you mean with "may not work" @tengqm?

Reply:
Why? it works for me, and it pins to a commit, so it won't change under normal circumstances.

Reply:
There is a syntax error, instead of `[]/healthz endpoint code in service.js](...)` It should be this 
```
[`/healthz` endpoint code in `service.js`](https://github.com/kubernauts/dok-example-us/blob/2334ee8fb11f8813370122bd46285cf45bdd4c48/stock-con/service.js#L52)
```

Reply:
Fixed